systems/controllers/endeffector_position_controller.cc

```diff
@@ -6,88 +6,104 @@ namespace systems{
-	  endpoint_orientation_commanded_port)->CopyToVector();
+	  endpoint_orientation_commanded_port_)->CopyToVector();
 
 
   Eigen::Vector3d x_actual;
-  const std::unique_ptr<Context<double>> plant_context = plant_.CreateDefaultContext();
+  const std::unique_ptr<Context<double>> plant_context =
+      plant_.CreateDefaultContext();
   plant_.SetPositions(plant_context.get(), q_actual);
-  plant_.CalcPointsPositions(*plant_context, ee_joint_frame, ee_contact_frame_,
-	                         plant_world_frame, &x_actual);
+  plant_.CalcPointsPositions(*plant_context, ee_joint_frame_, ee_contact_frame_,
+	                         plant_world_frame_, &x_actual);
 
-  MatrixXd diff = k_p * (x_desired - x_actual);
+  VectorXd diff = k_p_ * (x_desired - x_actual);
 
   // Quaternion for rotation from base to end effector
   Eigen::Quaternion<double> quat_n_a = plant_.CalcRelativeTransform(
-	  *plant_context, plant_world_frame, ee_joint_frame).rotation().ToQuaternion();
+	  *plant_context, plant_world_frame_, ee_joint_frame_).rotation().ToQuaternion();
 
   // Quaternion for rotation from world frame to desired end effector attitude.
   Eigen::Quaternion<double> quat_n_a_des = Eigen::Quaternion<double>(
 	  orientation_desired(0), orientation_desired(1), orientation_desired(2),
 	  orientation_desired(3));
 
-  // Quaternion for rotation from end effector attitude to desired end effector attitude.
-  Eigen::Quaternion<double> quat_a_a_des = quat_n_a.conjugate().operator*(quat_n_a_des);
+  // Quaternion for rotation
+  // from end effector attitude to desired end effector attitude.
+  Eigen::Quaternion<double> quat_a_a_des =
+      quat_n_a.conjugate().operator*(quat_n_a_des);
 
   // Angle Axis Representation for the given quaternion
-  Eigen::AngleAxis<double> angleaxis_a_a_des = Eigen::AngleAxis<double>(quat_a_a_des);
+  Eigen::AngleAxis<double> angleaxis_a_a_des =
+      Eigen::AngleAxis<double>(quat_a_a_des);
   MatrixXd axis = angleaxis_a_a_des.axis();
-  MatrixXd angularVelocity = k_omega * axis * angleaxis_a_a_des.angle();
+  MatrixXd angularVelocity = k_omega_ * axis * angleaxis_a_a_des.angle();
 
   // Transforming angular velocity from joint frame to world frame
-  MatrixXd angularVelocityWF = plant_.CalcRelativeTransform(
-	  *plant_context, ee_joint_frame, plant_world_frame).rotation() * angularVelocity;
+  VectorXd angularVelocityWF = plant_.CalcRelativeTransform(
+	  *plant_context, ee_joint_frame_, plant_world_frame_).rotation() * angularVelocity;
 
-  std::cout << x_desired << std::endl;
-  printf("@@@@@@@@@@@@@@@@@\n");
-  std::cout << x_actual << std::endl;
-
-  // Limit maximum commanded velocities
+  // Limit maximum commanded linear velocity
   double linear_speed_limit = 0.5;
-
-  double currVel = sqrt(diff(0, 0)*diff(0, 0) + diff(1, 0)*diff(1, 0) + diff(2, 0)*diff(2, 0));
+  double currVel = diff.norm();
 
   if (currVel > linear_speed_limit) {
       for (int i = 0; i < 3; i++) {
```

Comment:
You shouldn't need need to perform this rescaling in a for loop. 
`diff = diff * linear_speed_limit/currVel` should be sufficient